lib/nostrum/struct/channel.ex

```diff
@@ -37,132 +134,281 @@ defmodule Nostrum.Struct.Channel do
-  - `13` - `GUILD_STAGE_VOICE`    a voice channel for hosting events with an audience
+  @typedoc """
+  The video quality mode of the channel.
 
+  More information about _video quality modes_ can be found on the [Discord API Video Quality Mode Documentation](https://discord.com/developers/docs/resources/channel#channel-object-video-quality-modes).
   """
-  @type type :: integer()
+  @typedoc since: "0.5"
+  @type video_quality_mode :: integer() | nil
+
+  @typedoc """
+  Approximate count of messages in a thread, capped at 50.
+  """
+  @typedoc since: "0.5"
+  @type message_count :: integer() | nil
+
+  @typedoc """
+  Approximate count of members in a thread, capped at 50.
+  """
+  @typedoc since: "0.5"
+  @type member_count :: integer() | nil
+
+  @typedoc """
+  The video quality mode of the voice channel.
+
+  More information about _video quality modes_ can be found on the [Discord API Video Quality Mode Documentation](https://discord.com/developers/docs/resources/channel#channel-object-video-quality-modes).
+  """
+  @typedoc since: "0.5"
+  @type thread_metadata :: %{
+          archived: archived,
+          auto_archive_duration: auto_archive_duration,
+          archive_timestamp: archive_timestamp,
+          locked: boolean()
+        }
+
+  @typedoc """
+  When the thread was archived.
+
+  Timestamp per [ISO 8601:2019](https://en.wikipedia.org/wiki/ISO_8601). See also: `DateTime.from_iso8601/2`.
+  """
+  @typedoc since: "0.5"
+  @type archive_timestamp :: String.t() | nil
 
   @typedoc """
-  A `Nostrum.Struct.Channel` that represents a text channel in a guild
+  The threads locked status.
+  """
+  @typedoc since: "0.5"
+  @type locked :: boolean()
+
+  @typedoc """
+  The threads archived status.
+  """
+  @typedoc since: "0.5"
+  @type archived :: boolean()
+
+  @typedoc """
+  Archive duration for the thread.
+
+  - 60, 1 hour
+  - 1440, 24 hours
+  - 4320, 3 days
+  - 10080, 7 days
+  """
+  @typedoc since: "0.5"
+  @type auto_archive_duration :: integer()
+
+  @typedoc """
+  Present when the bot joins a thread.
+
+  Note: This is omitted on threads that the bot can immediately access on `:GUILD_CREATE` events received.
+  """
+  @typedoc since: "0.5"
+  @type member :: %{
+          id: id,
+          user_id: user_id,
+          join_timestamp: join_timestamp,
+          flags: flags
+        }
+
+  @typedoc """
+  User id of the threads creator.
+  """
+  @typedoc since: "0.5"
+  @type user_id :: Snowflake.t() | nil
+
+  @typedoc """
+  When the user joined the thread.
+
+  Timestamp per [ISO 8601:2019](https://en.wikipedia.org/wiki/ISO_8601). See also: `DateTime.from_iso8601/2`.
+  """
+  @typedoc since: "0.5"
+  @type join_timestamp :: String.t()
+
+  @typedoc """
+  User thread settings, currently only used for notifications.
+  """
+  @typedoc since: "0.5"
+  @type flags :: integer()
+
+  @typedoc """
+  Default duration for newly created threads.
```

Comment:
So these are in minutes? I think the example gets us there but we should just mention it IMO!